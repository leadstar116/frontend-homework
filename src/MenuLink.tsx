import { useDispatch } from 'react-redux'
import { toggleDropdownMenuVisibility } from './actions'

type Props = {
  text: string,
  icon: string,
}

const MenuLink = ({ icon, text }: Props) => {
  const dispatch = useDispatch();

  const handleMenuClick = () => {
    dispatch(toggleDropdownMenuVisibility());
  }

  return (
    <div className="menu-link-item menu-item" onClick={handleMenuClick}>
      <i className="material-icons">
        {icon}
      </i>

      {text}
    </div>
  );
};

export default MenuLink;
