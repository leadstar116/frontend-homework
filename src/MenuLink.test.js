import { shallow } from 'enzyme'
import MenuLink from './MenuLink'
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

const mockStore = configureStore();

describe('<MenuLink />', () => {
  let store = mockStore({
    selectedCompanyId: 1,
  });
  store.dispatch = jest.fn();

  it('renders link', () => {
    expect(shallow(
      <Provider store={store}>
        <MenuLink icon="phone" text="Call me" />
      </Provider>)
    ).toMatchSnapshot()
  })
})
