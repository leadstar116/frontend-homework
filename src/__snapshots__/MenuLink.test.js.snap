// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuLink /> renders link 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [MockFunction],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [MockFunction],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <MenuLink
    icon="phone"
    text="Call me"
  />
</ContextProvider>
`;
